x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/registry/ModelRegistry.java

```diff
@@ -224,11 +226,17 @@ public void clearDefaultIds() {
      * @throws ResourceNotFoundException if the specified id is guaranteed to not exist in the cluster.
      */
     public MinimalServiceSettings getMinimalServiceSettings(String inferenceEntityId) throws ResourceNotFoundException {
+        synchronized (this) {
+            assert lastClusterState != null : "initial cluster state not set yet";
```

Comment:
Nitpick: Redundant checks

Reply:
That's the hard failure mode for tests to catch issue early.